src/patternfly/components/DataList/data-list.scss

```diff
@@ -272,7 +268,7 @@
   margin-right: var(--pf-c-data-list__item-control--MarginRight);
 
   > *:not(:last-child) {
-    margin-right: var(--pf-c-data-list__item-control--not-last-child--MarginRight);
+    --pf-c-data-list__item-control--MarginRight: var(--pf-c-data-list__item-control--not-last-child--MarginRight);
```

Comment:
This removes the margin from children of `__item-control` since it just modifies the var for `__item-control`'s `margin-right`. It looks fine just to revert that.